sql/codegen.go

```diff
@@ -276,59 +259,98 @@ classifier = {{.Estimator}}(
 	{{end}}
 
 {{if .Train}}
-def train_input_fn(features, labels, batch_size):
-    dataset = tf.data.Dataset.from_tensor_slices((dict(features), labels))
-    dataset = dataset.shuffle(1000).repeat().batch(batch_size)
-    return dataset
+def train_input_fn(batch_size):
+	feature_types = dict()
+	feature_shapes = dict()
+	for name in feature_column_names:
+		if column_name_to_type[name] == "categorical_column_with_identity":
+			feature_types[name] = tf.int64
+			feature_shapes[name] = tf.TensorShape([None])
+		else:
+			feature_types[name] = tf.float32
+			feature_shapes[name] = tf.TensorShape([])
+
+	gen = db_generator(driver, conn, """{{.StandardSelect}}""",
+		feature_column_names, "{{.Y.Name}}", column_name_to_type)
+	dataset = tf.data.Dataset.from_generator(gen, (feature_types, tf.int64), (feature_shapes, tf.TensorShape([1])))
+	dataset = dataset.shuffle(1000).repeat().batch(batch_size)
```

Comment:
How are `.shuffle` and `.repeat` implemented if the whole dataset is not loaded in the memory?

Reply:
Answer: to achieve `.repeat`, `dataset` will call `gen` multiple times.